spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerMutationFactoryImpl.java

```diff
@@ -95,8 +95,9 @@ public Mutation update(Object object, Optional<Set<String>> includeColumns) {
 	}
 
 	@Override
+	@SuppressWarnings("unchecked")
 	public Mutation delete(Object object) {
-		return delete((Class<Object>) object.getClass(), Arrays.asList(object));
+		return delete((Class) object.getClass(), Collections.singletonList(object));
```

Comment:
Shouldn't the cast have a type for the generic? e.g., `Class<?>` or `Class<T>`?
Using generic types without the generic is a smell...

Reply:
Unfortunately casting to `Class<?>` will not work because it will not match with the signature of the `delete` function it's calling. 

Casting to `Class<T>` is also valid, but it does not solve the type warning. 

The issue seems to be `object.getClass()` just gives back an old fashioned `Class` without the T type param, and `delete` function called inside requires that `object.getClass()` give a Class with the T param that matches the type of `object` itself.

Reply:
done